### PR TITLE
add check for solver installation when initializing solver class instances

### PR DESCRIPTION
**Pull request by daniel-rdt**
This PR adds a check to the initialization of a solver instance to check whether the solver package is installed or not. An `ImportError` is raised if the necessary package is not installed.
fyi @FabianHofmann @lkstrp

### CODE REVIEW

**Review by lkstrp — COMMENTED** [review]
Can we not just use `available_solvers` instead of checking again?

https://github.com/PyPSA/linopy/blob/acac41476814f12e59b964c5d2cfcb03785dac27/linopy/solvers.py#L50-L115

**Comment by daniel-rdt** [comment]
We could even not specify the name but just one abstract function to get the name from the class __name__ via the SolverName enum:
```
def get_name_str(self):
    return SolverName[self.__class__.__name__].value
```
I'll made the changes

File: linopy/solvers.py [hunk 1]
```diff
@@ -209,6 +210,11 @@ def __init__(
     ):
         self.solver_options = solver_options
 
+        # Check for the solver to be initialized whether the package is installed or not.
+        if self.get_name_str() not in available_solvers:
```
[the remark this answers is not in the document]
Comment (daniel-rdt) [reply linopy/solvers.py:213]: I think I prefer the below version as it adds the option to access the SolverNames enum entry. We would need to specify here as:
```suggestion
        if self.solver_name.value not in available_solvers:
```

File: linopy/solvers.py [hunk 2]
```diff
@@ -209,6 +209,11 @@ def __init__(
     ):
         self.solver_options = solver_options
 
+        # Check for the solver to be initialized whether the package is installed or not.
+        if self.solver_name.value not in available_solvers:
+            msg = f"Solver package for '{self.get_name_str()}' is not installed. Please install first to initialize solver instance."
```
Comment (daniel-rdt) [inline linopy/solvers.py:214]:
```suggestion
            msg = f"Solver package for '{self.solver_name.value}' is not installed. Please install first to initialize solver instance."
```

**Pull request merged**